### PR TITLE
Additional Handshake check between proxy & its proxied validator

### LINKED ISSUE #1098: Proxy should only connect validator with configured signing key
### Expected Behavior

If a peer connects to the internal proxy port, but is not in possession of the key configured with the `--proxy.proxiedvalidatoraddress` flag, the peer should not receive forwarded messages, and should be dropped as a peer. An error message should be logged on the proxy.

### Current Behavior

Currently any node connected to the internal proxy port will receive forwarded consensus messages. No error will be seen on either the proxy or the peer which is attempting to act as a validator. In the case that this is a misconfiguration, as happened to a validator recently on mainnet, the validator will not be able to provide the proxy with the a valid enode certificate and as a result the node will remain partially connected to the validator network.

### Steps to Reproduce Behavior

Start a validator and proxy pair with all configurations correct except the `--proxy.proxiedvalidatoraddress` on the proxy. (It may be set to any valid address)

### System information

v1.0.0

### PR DESCRIPTION
**Pull request by YoduYodu**
### Description

**Notes**
This PR is mostly done, plan is to merge this after main net 1.2.0 release & after new protocol between proxy & validator has been established. See convo here: https://github.com/celo-org/celo-blockchain/pull/1168#pullrequestreview-512195104

**Expected Behavior**
If a peer connects to the internal proxy port, but is not in possession of the key configured with the --proxy.proxiedvalidatoraddress flag, the peer should not receive forwarded messages, and should be dropped as a peer. An error message should be logged on the proxy.

**Current Behavior**
Currently any node connected to the internal proxy port will receive forwarded consensus messages. No error will be seen on either the proxy or the peer which is attempting to act as a validator. In the case that this is a misconfiguration, as happened to a validator recently on mainnet, the validator will not be able to provide the proxy with the a valid enode certificate and as a result the node will remain partially connected to the validator network.

### Other changes

Using succinct Handshake message between proxy & proxied validator, instead of using `EnodeCertMsg`

### Tested

A testnet of 5 validators, one of which has 2 proxies:

**case 1:** 2 proxies set correct `proxy.proxiedvalidatoraddress` :white_check_mark:
Testnet is mining block with 5 validators
2 proxies register proxiedValidator

**case 2:** 1 correct, 1 incorrect: :white_check_mark:
Testnet is mining block with 5 validators
1 proxy registers proxiedValidator, while the other one doesn't

**case 3:** 2 incorrect: :white_check_mark:
Testnet is mining block with 4 validators
2 proxies don't register proxiedValidator

### Related issues

- Fixes #1098 

### Backwards compatibility

Yes

### CODE REVIEW

**Comment by kevjue** [comment]
> 
> **case 3:** 2 incorrect:
> Testnet is NOT mining block correctly ✅
> 2 proxies don't register proxiedValidator ✅
> 

Hmm.  I am a little surprised that mining stopped for this case.  4 out of 5 validators should be healthy (which is > 2/3 quorum), so in theory, block mining should of been ok.

**Review by kevjue — COMMENTED** [review]
This looks great!  A few notes;

1).  PTAL at this comment (https://github.com/celo-org/celo-blockchain/pull/1168/#issuecomment-712467740).  Your observation doesn't seem like it should be expected behavior.
2). I left a few comments. PTAL.
3). Let's merge this into master after the 1.2.0 release (tentative plan is the cut that release from master this coming friday).
4). There has been discussions of creating a separate network protocol for the proxy<->it's validator communication.  I think that when that is done, your changes should be part of it.

**Comment by YoduYodu** [comment]
> > **case 3:** 2 incorrect:
> > Testnet is NOT mining block correctly ✅
> > 2 proxies don't register proxiedValidator ✅
> 
> Hmm. I am a little surprised that mining stopped for this case. 4 out of 5 validators should be healthy (which is > 2/3 quorum), so in theory, block mining should of been ok.

Oh, the testnet is actually mining blocks with 4 validators engaged, I thought that was **incorrect**. Let me rephrase it.

File: consensus/istanbul/backend/handler.go [hunk 1]
```diff
@@ -339,6 +339,16 @@ func (sb *Backend) Handshake(peer consensus.Peer) (bool, error) {
 					msg = enodeCertMsg.Msg
 				}
 			}
+		} else if sb.IsProxiedValidator() {
```
[the remark this answers is not in the document]
Comment (YoduYodu) [reply consensus/istanbul/backend/handler.go:342]: I see, but will the signed message be used multiple times? 
I did observe multiple Handshakes between proxy & its proxied validator in the local testnet when `proxy.proxiedvalidatoraddress` is wrongly set, and the connection can be built nonetheless, and then they disconnect when proxied validator send msg to proxy, which couldn't pass certain checks within `proxy_engine`.

In other words, will this empty message be used in other scenarios?

**Review by 096889 — APPROVED** [review]

**Comment by YoduYodu** [comment]
Closed due to out of date
